components/gitpod-protocol/src/util/scrubbing.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { createHash } from "crypto";
+
+const redactedFields = ["auth_", "password", "token", "key", "jwt", "secret", "email"];
```

Comment:
do we actually ever want to log such values?

Reply:
I believe this gives a false sense of security

Reply:
I agree that on our side we can change all clients use stict API with prefefined list of properties, but then it can be worked around as well as don't apply to console.log which can be used by 3rd party dependencies. (i.e. strict api is going to give a false sense of security as well)

Reply:
Did we consider issuing an (additional) log message, probably `WARN`, that tells us that we logged one of the values here somewhere? :thinking:

Reply:
We could change to WARN if it helps for `redacted:nested`.